pkg/test/utils/kubernetes.go

```diff
@@ -372,16 +372,58 @@ func Namespaced(dClient discovery.DiscoveryInterface, obj runtime.Object, namesp
 	return m.GetName(), namespace, nil
 }
 
+func pruneLargeAdditions(expected *unstructured.Unstructured, actual *unstructured.Unstructured) runtime.Object {
+	pruned := actual.DeepCopy()
+	prune(expected.Object, pruned.Object)
+	return pruned
+}
+
+func prune(expected map[string]interface{}, actual map[string]interface{}) {
+	const maxLines = 10
```

Comment:
how did we come to 10 as maxLines?

Reply:
I tried to find something that's low enough to hide huge fields like `metadata.managedFields`, but large enough such that typical `metadata.labels` still shows since it might be useful for identifying reported objects like pods. We can always turn this into a knob later...
Actually let me add such comment here.